tests/Makefile

```diff
@@ -0,0 +1,107 @@
+# *
+# * Copyright 2023 RDK Management
+# *
+# * Licensed under the Apache License, Version 2.0 (the "License");
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# * http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+
+TARGET_BIN?= ut-control-test
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+NC='\033[0m'
+
+$(info $(shell echo -e ${GREEN}TARGET_DYNAMIC_LIB [$(TARGET_DYNAMIC_LIB)]${NC}))
+
+UT_CONTROL_TEST_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+export PATH := $(shell pwd)/toolchain:$(PATH)
+TOP_DIR ?= $(UT_CONTROL_TEST_DIR)
+BIN_DIR ?= $(TOP_DIR)/bin
+UT_DIR ?= $(UT_CONTROL_TEST_DIR)/ut-core
+
+# Enable CUnit Requirements
+CFLAGS += -DUT_CUNIT
+CUNIT_DIR +=  $(UT_CONTROL_TEST_DIR)/ut-core/framework/CUnit-2.1-3/CUnit
+CUNIT_SRC_DIRS += $(CUNIT_DIR)/Sources
+INC_DIRS += $(CUNIT_DIR)/Headers
```

Comment:
Something wrong with this makefile.

this module has no idea `CUNIT` exists, it only deals with `ut-core`

check the makefiles from a testing suite like `hdmi_cec_halif_test`